index.js

```diff
@@ -137,9 +138,27 @@ module.exports = {
         'semi': ['error', 'always', {
             omitLastInOneLineBlock: true
         }],
-        // Allow for multiple method signatures in TS (which appear as dupe class members)
-        "no-dupe-class-members": "off",
-        "no-redeclare": "off",
-        "@typescript-eslint/no-redeclare": ['error', {ignoreDeclarationMerge: true}]
-    }
+    },
+    // Run typescript-specific rules only on typescript files
+    overrides: [
+        {
+            'parserOptions': {'project': ['./tsconfig.json']},
+            // Note: Remember to disable the base rules when enabling the ts version
+            rules: {
+                'no-unused-vars': 'off',
+                '@typescript-eslint/no-unused-vars': ['error', {
+                    ignoreRestSiblings: true,
+                    args: 'none'
+                }],
+                // Sets 'return-await' to 'in-try-catch' mode
+                'no-return-await': 'off',
+                '@typescript-eslint/return-await': ['error', 'in-try-catch'],
+                // Allow for multiple method signatures in TS (which appear as dupe class members)
+                'no-dupe-class-members': 'off',
+                'no-redeclare': 'off',
+                '@typescript-eslint/no-redeclare': ['error', {ignoreDeclarationMerge: true}]
+            },
+            files: ['*.ts', '*.tsx'],
```

Comment:
Let's just move this to be the first key for this override object - will make its scope a bit more obvious